test/blocks/quiz/utils.test.js

```diff
@@ -0,0 +1,188 @@
+import { readFile } from '@web/test-runner-commands';
+import { expect } from '@esm-bundle/chai';
+import sinon from 'sinon';
+import { setConfig } from '../../../libs/utils/utils.js';
+import { userSelection, answers, resultRules, resultData } from './mocks/mock-states.js';
+
+const {
+  initConfigPathGlob, getQuizData, handleNext,
+  getAnalyticsDataForBtn, structuredFragments, nestedFragments,
+  getAnalyticsDataForLocalStorage, parseResultData,
+  getRedirectUrl, transformToFlowData, storeResultInLocalStorage, findMatchForSelections,
+  findAndStoreResultData,
+} = await import('../../../libs/blocks/quiz/utils.js');
+
+const locales = { '': { ietf: 'en-US', tk: 'hah7vzn.css' } };
+const conf = { locales };
+const QUIZ_BASE_PATH = 'https://main--milo--adobecom.hlx.page/delta/app-recommender/';
+
+setConfig(conf);
+
+let fetch; let quiz; let mockQuestionsData; let mockDataStrings;
+
+async function mockQuizResourceCall(mockFilePath, resourceType) {
+  const responseData = await readFile({ path: mockFilePath });
+  fetch.withArgs(`${QUIZ_BASE_PATH}${resourceType}.json`).resolves({ ok: true, json: () => JSON.parse(responseData) });
+}
+
+describe('Quiz', () => {
+  before(async () => {
+    fetch = sinon.stub(window, 'fetch');
+    document.body.innerHTML = await readFile({ path: './mocks/index.html' });
+    quiz = document.querySelector('.quiz');
+    initConfigPathGlob(quiz);
+    await mockQuizResourceCall('./mocks/questions.json', 'questions');
+    await mockQuizResourceCall('./mocks/strings.json', 'strings');
+    await mockQuizResourceCall('./mocks/results.json', 'results');
+    const [questions, dataStrings] = await getQuizData(quiz);
+    mockQuestionsData = questions;
+    mockDataStrings = dataStrings;
+  });
+
+  after(() => {
+    sinon.restore();
+  });
+
+  it('Checking config values from the quiz block', async () => {
+    const { configPath, quizKey, analyticsType, analyticsQuiz } = initConfigPathGlob(quiz);
+    expect(configPath).to.be.a('function');
+    expect(quizKey).to.be.a.string;
+    expect(analyticsType).to.be.a.string;
+    expect(analyticsQuiz).to.be.a.string;
+  });
+  it('Checking quiz data', async () => {
+    expect(mockQuestionsData).to.be.an('object');
+    expect(mockDataStrings).to.be.an('object');
+  });
+  it('Checking general next button functionality', async () => {
+    const selectedQuestion = { 'max-selections': '3', 'min-selections': '1', questions: 'q-category' };
+    const userInputSelections = { photo: true };
+
+    const { nextQuizViews } = handleNext(
+      mockQuestionsData,
+      selectedQuestion,
+      userInputSelections,
+      [],
+    );
+    expect(nextQuizViews).to.be.an('array');
+    expect(nextQuizViews.length).to.equal(2);
```

Comment:
I see L81 uses `expect(nextQuizViews).to.be.an('array').of.length(2);`, could the same not be applied here as well?

Reply:
Sure. Changed it.